### PR TITLE
Auth/Improvements

### PR DESCRIPTION
**Pull request by borut-t**
The last batch of updates.

### CODE REVIEW

File: Sources/Auth/Apple/AppleAuthenticator.swift [hunk 1]
```diff
@@ -11,20 +11,15 @@ import Foundation
...
-  private let userIdStorageKey = "povioKit.appleSocialProvider.signIn.userId"
-  private let storage: UserDefaults = .standard
+  private let userIdStorageKey = "signIn.userId"
+  private let storage: UserDefaults
   private let provider: ASAuthorizationAppleIDProvider
   private var processingPromise: Promise<Response>?
   
   public override init() {
     self.provider = .init()
+    self.storage = .init(suiteName: "povioKit.appleAuthenticator") ?? .standard
```
Comment (tonikocjan) [inline Sources/Auth/Apple/AppleAuthenticator.swift:22]: Should we allow the user to initialise the `storage`?

File: Sources/Auth/Core/SocialAuthenticator.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//
+//  Created by Borut Tomazin on 30/01/2023.
+//  Copyright © 2023 Povio Inc. All rights reserved.
+//
+
+import PovioKitPromise
+import Foundation
+import UIKit
+
+public class SocialAuthenticator {
```
Comment (tonikocjan) [inline Sources/Auth/Core/SocialAuthenticator.swift:13]: Maybe rename to smth like `SocialAuthenticationManager`?

File: Sources/Auth/Core/SocialAuthenticator.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  private let authenticators: [Authenticator]
+  
+  public init(authenticators: [Authenticator]) {
+    self.authenticators = authenticators
+  }
+}
+
+extension SocialAuthenticator {
```
Comment (tonikocjan) [inline Sources/Auth/Core/SocialAuthenticator.swift:21]: Should we handle URLs as well?

```swift
public func canOpenUrl(_ url: URL, application: UIApplication, options: [UIApplication.OpenURLOptionsKey: Any]) -> Bool {
  for auth in authenticators {
    if auth.canOpenUrl(url, application, options) { return true }
  }
  return false
}
```

**Review by tonikocjan — APPROVED** [review]
This looks great!

File: Sources/Auth/Core/SocialAuthenticator.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  public func authenticator<A: Authenticator>(for type: A.Type) -> A? {
+    authenticators.first { $0 is A } as? A
+  }
+  
+  public func signOut() {
```
Comment (tonikocjan) [inline Sources/Auth/Core/SocialAuthenticator.swift:26]: This should return a promise. Another implementation would be to just call `signOut` on all the authenticators.

File: Sources/Auth/Apple/AppleAuthenticator.swift (hunk 1 above)
Comment (borut-t) [reply Sources/Auth/Apple/AppleAuthenticator.swift:22]: I think we should yes.

File: Sources/Auth/Core/SocialAuthenticator.swift (hunk 4 above)
Comment (borut-t) [reply Sources/Auth/Core/SocialAuthenticator.swift:26]: Why?

**Review by markom-povio — APPROVED** [review]
👍

**Pull request merged**